src/dll/fuse/fuse_intf.c

```diff
@@ -1006,6 +1006,16 @@ static NTSTATUS fsp_fuse_intf_Open(FSP_FILE_SYSTEM *FileSystem,
         break;
     }
 
+    if (!(FileInfoBuf.FileAttributes & FILE_ATTRIBUTE_DIRECTORY) && (GrantedAccess & FILE_APPEND_DATA))
```

Comment:
My preference would be for this whole code block to be moved inside the else block that deals with files. I find that it makes clearer what is happening. Perhaps also add a comment to explain why this was necessary. The description that you used for this PR request should be sufficient.

Something along the lines of:

```C
    else
    {
        /*
         * Your PR description as comment.
         */
        if (GrantedAccess & FILE_APPEND_DATA)
        {
            // ...
        }

        if (0 != f->ops.open)
        {
            err = f->ops.open(contexthdr->PosixPath, &fi);
            Result = fsp_fuse_ntstatus_from_errno(f->env, err);
        }
    }
```

Reply:
No problem, will do.

Reply:
Thank you!